--- a/arch/mips/ralink/mt7620.c	2020-04-24 18:33:37.699033995 +0000
+++ b/arch/mips/ralink/mt7620.c	2020-04-24 20:53:53.000000000 +0000
@@ -185,6 +185,31 @@
 	FUNC("gpio", 0, 11, 1),
 };
 
+static struct rt2880_pmx_func ephy_p1_grp_mt7628[] = {
+    FUNC("rsvd", 3, 42, 1),
+    FUNC("rsvd", 2, 42, 1),
+    FUNC("gpio", 1, 42, 1),
+    FUNC("ephy", 0, 42, 1),
+};
+static struct rt2880_pmx_func ephy_p2_grp_mt7628[] = {
+    FUNC("rsvd", 3, 41, 1),
+    FUNC("rsvd", 2, 41, 1),
+    FUNC("gpio", 1, 41, 1),
+    FUNC("ephy", 0, 41, 1),
+};
+static struct rt2880_pmx_func ephy_p3_grp_mt7628[] = {
+    FUNC("rsvd", 3, 40, 1),
+    FUNC("rsvd", 2, 40, 1),
+    FUNC("gpio", 1, 40, 1),
+    FUNC("ephy", 0, 40, 1),
+};
+static struct rt2880_pmx_func ephy_p4_grp_mt7628[] = {
+    FUNC("rsvd", 3, 39, 1),
+    FUNC("rsvd", 2, 39, 1),
+    FUNC("gpio", 1, 39, 1),
+    FUNC("ephy", 0, 39, 1),
+};
+
 static struct rt2880_pmx_func wled_kn_grp_mt7628[] = {
 	FUNC("rsvd", 3, 35, 1),
 	FUNC("rsvd", 2, 35, 1),
@@ -202,6 +227,10 @@
 #define MT7628_GPIO_MODE_MASK		0x3
 
 #define MT7628_GPIO_MODE_WLED_KN	48
+#define MT7628_GPIO_MODE_EPHY_P4	42
+#define MT7628_GPIO_MODE_EPHY_P3	40
+#define MT7628_GPIO_MODE_EPHY_P2	38
+#define MT7628_GPIO_MODE_EPHY_P1	36
 #define MT7628_GPIO_MODE_WLED_AN	32
 #define MT7628_GPIO_MODE_PWM1		30
 #define MT7628_GPIO_MODE_PWM0		28
@@ -237,6 +266,10 @@
 	GRP_G("gpio", gpio_grp_mt7628, MT7628_GPIO_MODE_MASK, 1, MT7628_GPIO_MODE_GPIO),
 	GRP_G("wled_an", wled_an_grp_mt7628, MT7628_GPIO_MODE_MASK, 1, MT7628_GPIO_MODE_WLED_AN),
 	GRP_G("wled_kn", wled_kn_grp_mt7628, MT7628_GPIO_MODE_MASK, 1, MT7628_GPIO_MODE_WLED_KN),
+	GRP_G("ephy_p1", ephy_p1_grp_mt7628, MT7628_GPIO_MODE_MASK, 1, MT7628_GPIO_MODE_EPHY_P1),
+    GRP_G("ephy_p2", ephy_p2_grp_mt7628, MT7628_GPIO_MODE_MASK, 1, MT7628_GPIO_MODE_EPHY_P2),
+    GRP_G("ephy_p3", ephy_p3_grp_mt7628, MT7628_GPIO_MODE_MASK, 1, MT7628_GPIO_MODE_EPHY_P3),
+    GRP_G("ephy_p4", ephy_p4_grp_mt7628, MT7628_GPIO_MODE_MASK, 1, MT7628_GPIO_MODE_EPHY_P4),
 	{ 0 }
 };
 
